src/monster.h

```diff
@@ -407,7 +407,7 @@ class monster : public Creature, public JsonSerializer, public JsonDeserializer
 
     private:
         int hp;
-        std::unordered_map<std::string, mon_special_attack> special_attacks;
+        std::map<std::string, mon_special_attack> special_attacks;
```

Comment:
PR looks fined. But why this change? Does it need to be ordered?

Also: can't you remove the old `mattack::leap` function and  convert the JSON to use the new function? To reduce code duplication and so.